src/Carbon/PHPStan/MacroScanner.php

```diff
@@ -29,8 +45,16 @@ final class MacroScanner
      */
     public function hasMethod(string $className, string $methodName): bool
     {
-        return is_a($className, CarbonInterface::class, true) &&
-            \is_callable([$className, 'hasMacro']) &&
+        $classReflection = $this->reflectionProvider->getClass($className);
+
+        if (
+            $classReflection->getName() !== CarbonInterface::class &&
+            !$classReflection->isSubclassOf(CarbonInterface::class)
+        ) {
+            return false;
+        }
+
+        return \is_callable([$className, 'hasMacro']) &&
```

Comment:
This should probably be `$classReflection->hasMethod('hasMacro')` but I will first check and merge this and check in a next step if this part can also be static.

Reply:
I think the `is_callable()` check could probably be removed, because `hasMacro` is defined on `CarbonInterface`. So we're already sure its callable (unless I'm missing something)

Reply:
It can be an old or modified version of `CarbonInterface` so it does not have `hasMacro` but yeah it's very unlikely.